src/cloudabi-run/yaml_file_descriptor_factory.h

```diff
@@ -32,11 +32,15 @@ class YAMLFileDescriptorFactory
                           std::vector<const argdata_t *> keys,
                           std::vector<const argdata_t *> values) override;
 
+  void SetExecutableFd(int fd);
+
  private:
   YAMLFactory<const argdata_t *> *const fallback_;
 
   std::vector<std::unique_ptr<argdata_t>> argdatas_;
   std::vector<std::shared_ptr<arpc::FileDescriptor>> fds_;
+
+  int execfd = -1;
```

Comment:
This can be marked constant. Please place it next to the other constant members of this class (`fallback_`) and add a trailing underscore to its name for consistency.